metrics/web/lib/debug_menu/presentation/state/debug_menu_notifier.dart

```diff
@@ -0,0 +1,163 @@
+  /// Otherwise, returns `false`.
+  bool get isLoading => _isLoading;
+
+  /// Returns `true` if current [LocalConfig] is not `null`.
+  /// Otherwise, returns `false`.
+  bool get isInitialized => _localConfig != null;
+
+  /// A view model that provides the [LocalConfig] data
+  /// for the FPS monitor feature.
+  FpsMonitorLocalConfigViewModel get fpsMonitorLocalConfigViewModel =>
+      _fpsMonitorLocalConfigViewModel;
+
+  /// Provides an error description that occurred
+  /// during updating the [LocalConfig].
+  String get localConfigUpdatingErrorMessage =>
+      _localConfigUpdatingError?.message;
+
+  /// Creates a new instance of the [DebugMenuNotifier]
+  /// with the given parameters.
+  ///
+  /// Throws an [AssertionError] if any of the given parameters is `null`.
+  DebugMenuNotifier(
+    this._openLocalConfigStorageUseCase,
+    this._readLocalConfigUseCase,
+    this._updateLocalConfigUseCase,
+    this._closeLocalConfigStorageUseCase,
+  )   : assert(_openLocalConfigStorageUseCase != null),
+        assert(_readLocalConfigUseCase != null),
+        assert(_updateLocalConfigUseCase != null),
+        assert(_closeLocalConfigStorageUseCase != null);
+
+  /// Initializes the [LocalConfig].
+  Future<void> initializeLocalConfig() async {
+    try {
+      _setIsLoading(true);
+
+      await _openLocalConfigStorageUseCase();
+
+      final config = _readLocalConfigUseCase();
+
+      _setLocalConfig(config);
+
+      _setIsLoading(false);
+    } catch (_) {
+      initializeDefaults();
+    }
+  }
+
+  /// Initializes the [LocalConfig] with default values.
+  void initializeDefaults() {
+    _setLocalConfig(
+      const LocalConfig(isFpsMonitorEnabled: false),
+    );
+
+    _setIsLoading(false);
+  }
+
+  /// Toggles the fps monitor feature and updates the [LocalConfig].
+  Future<void> toggleFpsMonitor() async {
+    final isFpsMonitorEnabled = _fpsMonitorLocalConfigViewModel.isEnabled;
+
+    final configParam = LocalConfigParam(
+      isFpsMonitorEnabled: !isFpsMonitorEnabled,
+    );
+
+    _resetLocalConfigUpdatingError();
+
+    try {
+      final newConfig = await _updateLocalConfigUseCase(configParam);
+
+      _setLocalConfig(newConfig);
+
+      notifyListeners();
+    } on PersistentStoreException catch (exception) {
+      _localConfigUpdatingErrorHandler(exception.code);
+    }
+  }
+
+  /// Sets the current [_localConfig] value to the given [config] and updates
+  /// the [_fpsMonitorLocalConfigViewModel].
+  void _setLocalConfig(LocalConfig config) {
+    _localConfig = config;
+
+    _fpsMonitorLocalConfigViewModel = FpsMonitorLocalConfigViewModel(
+      isEnabled: config.isFpsMonitorEnabled,
+    );
+  }
+
+  /// Sets the current [_isLoading] value to the given [value]
+  /// and notifies listeners.
+  void _setIsLoading(bool value) {
+    _isLoading = value;
+    notifyListeners();
+  }
+
+  /// Handles an error occurred during updating the [LocalConfig].
```

Comment:
Currently, it handles the code of the exception but not the exception itself. Should we rethink this moment?